### PR TITLE
add CalendarFieldAggregation

### PR DESCRIPTION
**Pull request by joeferner**
- [x] @sfeng88 
- [x] @srfarley 
- [x] @jharwig

### CODE REVIEW

File: core/src/main/java/org/vertexium/query/DefaultGraphQueryIterableWithAggregations.java [hunk 1]
```diff
@@ -55,7 +62,30 @@ public static boolean isAggregationSupported(Aggregation agg) {
...
+
+    private HistogramResult getCalendarFieldHistogramResult(final CalendarFieldAggregation agg, Iterator<T> it) {
+        String propertyName = agg.getPropertyName();
+        final Calendar calendar = GregorianCalendar.getInstance(agg.getTimeZone());
+        Map<Integer, List<T>> elementsByProperty = getElementsByProperty(it, propertyName, new ValueConverter<Integer>() {
+            @Override
+            public Integer convert(Object o) {
+                Date d = (Date) o;
+                calendar.setTime(d);
+                //noinspection MagicConstant
```
Comment (srfarley) [inline core/src/main/java/org/vertexium/query/DefaultGraphQueryIterableWithAggregations.java]: Why is this needed? There are no constants referenced on this line.

Comment (joeferner) [inline core/src/main/java/org/vertexium/query/DefaultGraphQueryIterableWithAggregations.java]: IntelliJ complains that there ISN'T a constant referenced on this line. It wants it to be a `Calendar` magic constant.
